### DISCUSSION TITLE
What is the default firewall for Github Actions runners

### DISCUSSION THREAD

**Discussion by gpotter2** [community/community#26184] [Actions]
<p>Hi,</p>
<p>I’m trying to port our <a href="https://github.com/secdev/scapy" rel="nofollow noopener">scapy</a> unit tests from Travis to Github actions.</p>
<p>While <a href="https://github.com/gpotter2/scapy/runs/476836553?check_suite_focus=true" rel="nofollow noopener">I’m quite successful at making it run</a>, some networking tests simply fail.</p>
<p>For instance, we try to ping <a href="http://www.google.com" rel="nofollow noopener">www.google.com</a> using PF_INET low-level sockets. We open the sockets ourselves (they need root), craft the whole frame from Ethernet to ICMP and send/receive it. (the tests do receive packets, but not the ICMP answers). All of this worked on Travis VMs. I assume ICMP pings are blocked</p>
<p>If I understood the doc correctly, Github actions is running on <a href="https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series" rel="nofollow noopener">https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series</a> machines. <strong>Are there restrictions when it comes to networking on those ?</strong></p>
<p><strong>Is there an external firewall like on Azure VMs ? if so is there a way to disable it/change it ?</strong></p>
<p>I tried to disable ufw and flush the iptables, to no avail</p>
<p>Thanks for your help</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/gpotter2">@gpotter2</a> ,</p>
<p>GitHub hosts Linux and Windows runners on Standard_DS2_v2 virtual machines in Microsoft Azure with the GitHub Actions runner application installed. And due to security policy Azure blocks ICMP by default. Hence, you cannot get ICMP answer in workflow.</p>
<p>As an alternative, you can try local <a href="https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" rel="nofollow noopener">self-hosted runner</a> instead.</p>
<p>Thanks.</p>
